content/renderer/render_frame_impl.cc

```diff
@@ -2580,6 +2586,16 @@ void RenderFrameImpl::CommitNavigation(
   DCHECK(!NavigationTypeUtils::IsSameDocument(common_params->navigation_type));
   LogCommitHistograms(commit_params->commit_sent, is_main_frame_);
 
+  absl::optional<recordreplay::AutoDependencyExecution> execute;
+  if (recordreplay::IsReplaying() && !recordreplay::FeatureEnabled("no-dependency-graph")) {
```

Comment:
I feel very strongly about just creating a central function for this check.
The biggest reason here is that this runtime correctness check (which triggers if the string has a typo) becomes a compile time check. You don't have to worry whether or not you copy-and-pasted this condition correctly.

Yes, currently, only you are working on this, but soon enough other people will.

Reply:
Fair enough, these feature enabled checks are temporary but a helper function will improve readability / greppability in the longer term as well.